### PR TITLE
Opd hmi15 report fix

### PR DESCRIPTION
**Pull request by dominickasanga**
- a fix hmi15 conditions, which was not capturing all
- added and endpont for caputing opd attendance, based on encounters done in opd

### CODE REVIEW

File: app/controllers/api/v1/reports_controller.rb [hunk 1]
```diff
@@ -47,6 +47,14 @@ def registration
     render json: stats
   end
 
+  def attendance
+    start_date, end_date = params.require %i[start_date end_date]
+    puts( params.require %i[start_date end_date])
```
Comment (Skipper-116) [inline app/controllers/api/v1/reports_controller.rb]: This is not necessary in production... Can we utilize the ProgramReportsController#Show (action method)

File: app/services/opd_service/report_engine.rb [hunk 2]
```diff
@@ -182,6 +182,20 @@ def registration(start_date, end_date)
       return stats
     end
 
+    def attendance(start_date, end_date)
```
Comment (Skipper-116) [inline app/services/opd_service/report_engine.rb]: Add this under REPORTS in OPDService::ReportEngine then add a class or module that handles this

File: app/services/report_service.rb [hunk 3]
```diff
@@ -57,6 +57,10 @@ def registration(start_date, end_date)
     engine(@program).registration(start_date, end_date)
   end
 
+  def attendance(start_date, end_date)
+    engine(@program).attendance(start_date, end_date)
```
Comment (Skipper-116) [inline app/services/report_service.rb]: This will not be necessary after you add those on recommendation on top

File: config/routes.rb [hunk 4]
```diff
@@ -307,6 +307,7 @@ def paginate_url(url, params)
   # OPD reports
   get '/api/v1/malaria_report' => 'api/v1/reports#malaria_report'
   get '/api/v1/registration' => 'api/v1/reports#registration'
+  get '/api/v1/attendance' => 'api/v1/reports#attendance'
```
Comment (Skipper-116) [inline config/routes.rb]: This will not be required after you add those on top

**Review by Skipper-116 — CHANGES_REQUESTED** [review]

File: .~lock.migration_scores_20220915.csv# [hunk 5]
```diff
@@ -0,0 +1 @@
+,dominic,Dom-Stig,15.09.2022 18:24,file:///home/dominic/.config/libreoffice/4;
```
Comment (Skipper-116) [inline .~lock.migration_scores_20220915.csv#]: This file is part of the report?

**Review by Skipper-116 — APPROVED** [review]

File: .~lock.migration_scores_20220915.csv# (hunk 5 above)
Comment (dominickasanga) [reply .~lock.migration_scores_20220915.csv#]: no.. let me remove it

Comment (dominickasanga) [reply .~lock.migration_scores_20220915.csv#]: no.. its not

**Pull request merged**